gamestonk_terminal/cryptocurrency/discovery/discovery_controller.py

```diff
@@ -0,0 +1,232 @@
+        self._discovery_parser.add_argument("cmd", choices=self.CHOICES)
+
+    def print_help(self):
+        """Print help"""
+        print("\nDiscovery:")
+        print("   cls             clear screen")
+        print("   ?/help          show this menu again")
+        print("   q               quit this menu, and shows back to main menu")
+        print("   quit            quit to abandon program")
+        print("")
+
+        print("CoinGecko:")
+        print("   cg_coins           coins available on CoinGecko")
+        print("   cg_trending        trending coins on CoinGecko")
+        print("   cg_most_voted      most voted coins on CoinGecko")
+        print("   cg_most_visited    most visited coins on CoinGecko")
+        print("   cg_top_volume      coins with highest volume on CoinGecko")
+        print("   cg_recently        recently added on CoinGecko")
+        print("   cg_sentiment       coins with most positive sentiment")
+        print(
+            "   cg_gainers         top gainers - coins which price gained the most in given period "
+        )
+        print(
+            "   cg_losers          top losers - coins which price dropped the most in given period "
+        )
+        print("   cg_yfarms          top yield farms")
+        print("   cg_top_defi        top defi protocols")
+        print("   cg_top_dex         top decentralized exchanges")
+        print("   cg_top_nft         top non fungible tokens")
+        print("")
+        print("CoinPaprika:")
+        print("   cp_coins           coins available on CoinPaprika")
+        print("   cp_search          search on CoinPaprika")
+        print("")
+        print("CoinMarket Cap:")
+        print("   cmc_top            top coins from CoinMarketCap")
+        print("")
+        print("Binance:")
+        print("   bin_coins          coins available on Binance")
+
+        print("")
+
+    def switch(self, an_input: str):
+        """Process and dispatch input
+
+        Returns
+        -------
+        True, False or None
+            False - quit the menu
+            True - quit the program
+            None - continue in the menu
+        """
+
+        # Empty command
+        if not an_input:
+            print("")
+            return None
+
+        (known_args, other_args) = self._discovery_parser.parse_known_args(
+            an_input.split()
+        )
+
+        # Help menu again
+        if known_args.cmd == "?":
+            self.print_help()
+            return None
+
+        # Clear screen
+        if known_args.cmd == "cls":
+            os.system("cls||clear")
+            return None
+
+        return getattr(
+            self, "call_" + known_args.cmd, lambda: "Command not recognized!"
+        )(other_args)
+
+    def call_help(self, _):
+        """Process Help command"""
+        self.print_help()
+
+    def call_q(self, _):
+        """Process Q command - quit the menu."""
+        print("Moving back to (crypto) menu")
```

Comment:
Probably this print is not needed as the terminal relies on this q to jump back 1 layer functionality